### PR TITLE
In ImageDataGenerator.flow shuffle before every epoch

### PR DESCRIPTION
**Pull request by udibr**
https://github.com/fchollet/keras/issues/1551

### CODE REVIEW

File: keras/preprocessing/image.py [hunk 1]
```diff
@@ -142,18 +142,21 @@ def __init__(self,
...
+            if b == 0:
+                if seed is not None:
+                    random.seed(seed + total_b)
+
+                if shuffle:
+                    seed = random.randint(1, 10e6)
+                    np.random.seed(seed)
+                    np.random.shuffle(X)
+                    np.random.seed(seed)
+                    np.random.shuffle(y)
```
Comment (fchollet) [inline keras/preprocessing/image.py]: More elegant would be to shuffle an array of indices once, then use the array to index X and y.

Comment (udibr) [inline keras/preprocessing/image.py]: fixed - the first draft was even a bug not just ugly

**Comment by udibr** [comment]
Error in Travis does not look related to me

**Comment by fchollet** [comment]
Looks good to me. Please squash your commits and I'll merge.

**Pull request merged**